streaming/base/shared/memory.py

```diff
@@ -47,28 +39,29 @@ def __init__(self,
         try:
             if create is True:
                 # Creates a new shared memory block
-                shm = SharedMemory(name, create, size)
+                shm = BuiltinSharedMemory(name, create, size)
                 self.created_shms.append(shm)
             elif create is False:
                 # Avoid tracking shared memory resources in a process who attaches to an existing
                 # shared memory block because the process who created the shared memory is
                 # responsible for destroying the shared memory block.
                 resource_tracker.register = self.fix_register
                 # Attaches to an existing shared memory block
-                shm = SharedMemory(name, create, size)
+                shm = BuiltinSharedMemory(name, create, size)
                 self.opened_shms.append(shm)
             else:
                 try:
                     # Creates a new shared memory block
-                    shm = SharedMemory(name, True, size)
+                    shm = BuiltinSharedMemory(name, True, size)
                     self.created_shms.append(shm)
                 except FileExistsError:
                     sleep(TICK)
                     resource_tracker.register = self.fix_register
                     # Attaches to an existing shared memory block
-                    shm = SharedMemory(name, False, size)
+                    shm = BuiltinSharedMemory(name, False, size)
                     self.opened_shms.append(shm)
             self.shm = shm
+            self.buf = shm.buf
```

Comment:
`SharedMemory` buf is of type `MemoryView` and it is not-pickable when using `Spawn` multiprocessing. Instead of saving `buf` as a class object, create a method which returns a `buf` object. Something like this

```
@property
def buf(self):
    return self.shm.buf
```